pkg/commands/compute/init.go

```diff
@@ -196,25 +211,104 @@ func (c *InitCommand) Exec(in io.Reader, out io.Writer) (err error) {
 		c.cloneFrom = from
 	}
 
-	// We only want to fetch a remote package if c.cloneFrom has been set.
-	// This can happen in two ways:
+	// There are three situations in which we might fetch something
+	// here. We might fetch a template if:
 	//
-	// 1. --from flag is set
+	// 1. --from flag is set to a template repository, or
 	// 2. user selects starter kit when prompted
 	//
+	// Or we fetch an existing, deployed package if
+	//
+	// 3. --from flag is set to a serviceID
+	//
 	// We don't fetch if the user has indicated their language of choice is
 	// "other" because this means they intend on handling the compilation of code
 	// that isn't natively supported by the platform.
 	if c.cloneFrom != "" {
-		err = c.FetchPackageTemplate(branch, tag, file.Archives, spinner, out)
-		if err != nil {
-			c.Globals.ErrLog.AddWithContext(err, map[string]any{
-				"From":      from,
-				"Branch":    branch,
-				"Tag":       tag,
-				"Directory": c.dir,
+		if !isExistingService {
+			err = c.FetchPackageTemplate(branch, tag, file.Archives, spinner, out)
+			if err != nil {
+				c.Globals.ErrLog.AddWithContext(err, map[string]any{
+					"From":      from,
+					"Branch":    branch,
+					"Tag":       tag,
+					"Directory": c.dir,
+				})
+				return err
+			}
+		} else {
+			profileName, _, _ := c.Globals.Profile()
```

Comment:
This presumes a profile is being used.

For example, I have profiles defined but I typically have them overridden by the `FASTLY_API_TOKEN` environment variable. See the following logic flow for more details: https://github.com/fastly/cli/blob/6c1ec982475e698bd347c94622f857d9e4f44c90/pkg/global/global.go#L125-L133

...basically it breaks down to...

- `--token` returns `lookup.SourceFlag`
- `FASTLY_API_TOKEN` returns `lookup.SourceEnvironment`
- Everything else (i.e. anything to do with a profile) returns `lookup.SourceFile`
- If no token found it returns `lookup.SourceUndefined`

Reply:
Is it OK to leave `profile` in `fastly.toml` blank? (I'm not really sure how it's used since profiles are defined locally outside of whatever project repo)

Reply:
`profile` is part of the CLI config and not the fastly.toml (the latter being a project-specific configuration file, where as the CLI's internal configuration file is used to store profile information: https://github.com/fastly/cli/blob/main/pkg/config/config.go#L199-L200)

Reply:
It is [included in the manifest file](https://github.com/fastly/cli/blob/main/pkg/manifest/file.go#L31-L32), though I'm not 100% sure why. If we can leave it blank there, I'll drop it from this completely, which seems safest

Reply:
I don't really recall to be honest but I have a suspicion that _maybe_ it's related to data collection?

Reply:
I took it out with no obvious negative effects. We could follow up with a prompt to include it if we find it's valuable